gcloud/resource_manager/project.py

```diff
@@ -0,0 +1,333 @@
+        from a Resource Manager client.
+
+    To create a new project::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.project('purple-spaceship-123')
+        >>> project.name = 'Purple Spaceship Project!'
+        >>> project.create()
+
+    To get an existing project::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.get_project('purple-spaceship-123')
+        >>> print project.name
+        Purple Spaceship Project!
+
+    To manage labels::
+
+        >>> from gcloud import resource_manager
+        >>> client = resource_manager.Client()
+        >>> project = client.get_project('purple-spaceship-123')
+        >>> project.labels = {'color': 'purple'}
+        >>> project.labels['environment'] = 'production'
+        >>> project.update()
+
+    See:
+    https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects
+
+    :type client: :class:`gcloud.resource_manager.client.Client`
+    :param client: The Client used with this project.
+
+    :type project_id: string
+    :param project_id: The globally unique ID of the project.
+
+    :type name: string
+    :param name: The name of the project.
+
+    :type labels: dict
+    :param labels: A list of labels associated with the project.
+    """
+    def __init__(self, client, project_id, name=None, labels=None):
+        self.client = client
+        self.project_id = project_id
+        self.name = name or None
+        self.number = None
+        self.labels = labels or {}
+        self.status = None
+
+    def __repr__(self):
+        return '<Project: %r (%r)>' % (self.name, self.project_id)
+
+    @classmethod
+    def from_api_repr(cls, resource, client):
+        """Factory:  construct a project given its API representation.
+
+        :type resource: dict
+        :param resource: project resource representation returned from the API
+
+        :type client: :class:`gcloud.resource_manager.client.Client`
+        :param client: The Client used with this project.
+
+        :rtype: :class:`gcloud.resource_manager.project.Project`
+        """
+        project = cls(project_id=resource['projectId'], client=client)
+        project.set_properties_from_api_repr(resource)
+        return project
+
+    def set_properties_from_api_repr(self, resource):
+        """Update specific properties from its API representation."""
+        self.name = resource.get('name')
+        self.number = resource['projectNumber']
+        self.labels = resource.get('labels', {})
+        self.status = resource['lifecycleState']
+
+    @property
+    def full_name(self):
+        """Fully-qualified name (ie, ``'projects/purple-spaceship-123'``)."""
+        if not self.project_id:
+            raise ValueError('Missing project ID.')
+        return 'projects/%s' % (self.project_id)
```

Comment:
The `(`, `)` do nothing here. Make it a tuple `(self.project_id,)` for best results.